src/ray/rpc/gcs_server/gcs_rpc_client.h

```diff
@@ -66,11 +72,25 @@ class Executor {
         executor->Retry();                                                             \
       }                                                                                \
     };                                                                                 \
-    auto operation = [request, operation_callback](GcsRpcClient *gcs_rpc_client) {     \
+    auto operation = [request, operation_callback,                                     \
+                      timeout_ms](GcsRpcClient *gcs_rpc_client) {                      \
       RAY_UNUSED(INVOKE_RPC_CALL(SERVICE, METHOD, request, operation_callback,         \
-                                 gcs_rpc_client->grpc_client, method_timeout_ms));     \
+                                 gcs_rpc_client->grpc_client, timeout_ms));            \
     };                                                                                 \
     executor->Execute(operation);                                                      \
+  }                                                                                    \
+                                                                                       \
+  ray::Status Sync_##METHOD(const METHOD##Request &request, METHOD##Reply *reply_in,   \
+                            const int64_t timeout_ms = method_timeout_ms) {            \
+    auto promise = std::make_shared<std::promise<Status>>();                           \
```

Comment:
promise doesn't need to be shared I think. Check the example [here](https://sourcegraph.com/github.com/ray-project/ray/-/blob/src/ray/core_worker/core_worker_process.cc?L182:29)

Reply:
Oh yeah. I think we needed it in the past, but not anymore with the timeout. (In the past it could be invoked after timeout happens if we have get_future.wait_for

Reply:
Let me try fixing jt